skiko/src/awtMain/kotlin/org/jetbrains/skiko/windows/JumpList.kt

```diff
@@ -0,0 +1,156 @@
+package org.jetbrains.skiko.windows
+
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import org.jetbrains.skiko.hostOs
+
+/**
+ * Provides access to Windows Jump List features
+ */
+object JumpList {
+    /**
+     * Returns a value that indicates whether Jump Lists are supported.
+     */
+    fun isSupported(): Boolean = when {
+        hostOs.isWindows -> true
+        else -> false
+    }
+
+    /**
+     * Wraps a single Jump List building transaction.
+     */
+    fun build(block: JumpListBuilder.() -> Unit) = when {
+        hostOs.isWindows -> JumpListBuilder().use { builder ->
+            builder.initialize()
+            block(builder)
+            builder.commit()
+        }
+        else -> error("Jump List is only supported on Windows")
+    }
+}
+
+/**
+ * Represents an item in the Jump List.
+ */
+data class JumpListItem(val title: String, val arguments: String) {
+    var attributes: JumpListItemAttributes? = null
+}
+
+/**
+ * Represents optional attributes that can be added to a Jump List item.
+ */
+class JumpListItemAttributes {
+    var description: String? = null
+    var icon: JumpListItemIcon? = null
+}
+
+/**
+ * Represents an icon that can be assigned to a Jump List item.
+ */
+data class JumpListItemIcon(val path: String, val index: Int) {
+    companion object {
+        internal fun fromParts(path: String?, num: Int): JumpListItemIcon? {
+            if (path.isNullOrEmpty()) return null
+            return JumpListItemIcon(path, num)
+        }
+    }
+}
+
+/**
+ * Provides functions to build the Jump List.
+ *
+ * This is a wrapper for ICustomDestinationList Windows COM interface.
+ */
+class JumpListBuilder internal constructor() : AutoCloseable {
```

Comment:
We can add items, but I didn't find a way how we can react on clicks on these items. Will it be added in a separate PR?

Reply:
No, I don't think there can be a single source of truth for reacting to the items. There's no "event" or message for this from what I know; the user must pass the "arguments" (see `JumpListItem`, maybe we should explicitly check that the arguments are not an empty string), and the click will launch a new instance of the app with the arguments. That's essentially the only way to react to it.

Reply:
What will be called when we clicked on it?

Reply:
The app that created it + arguments. See:
https://github.com/JetBrains/skiko/blob/9a126efbab1e75f616ebe14fb1429c94a72336b4/skiko/src/awtMain/cpp/windows/JumpList.cc#L107-L111

Reply:
Understood. I confused it with a task list. The current functionality is for recent files that creates a new process every time.

Reply:
In this case it's the same for us. We do pretty much the same thing as the UWP API, because the system provided "recents" category requires `IShellItem`s, but they have a number of limitations. For example, your app must be a registered handler for the file type, and no app except Explorer can add folders to the recents lists (and that's exactly what we need in Fleet). So we create `IShellLink`s instead which are commonly used for tasks but can also go into custom categories (so yeah, for now to have a Recents list you have to define your own category).